### PR TITLE
Calypso UI Components: DateRange: Refactor "header" and "footer" areas of component

### PR DESCRIPTION
**Pull request by annacmc**
Related to https://github.com/Automattic/red-team/issues/178

* In order to move the buttons (the sole occupant of the header component) to the footer area, I refactored to add a new footer component for the buttons, and move the existing title/header info into the header component. 
* This also involved adding a new `renderFooter` function + adding this to the docs

## Why are these changes being made?

* To extend the Calypso DateRange picker component for use with the Stats Date Control update.

## Testing Instructions

* open Calypso live branch
* navigate to `/devdocs/design/date-range`
* check the `default` demo at the top. Check that everything still displays correctly, including when selecting custom dates check that the `please select the first day` text changes to `please select the last day` to `reset` as you select date ranges
* check that the `cancel` + `apply` buttons display below the calendars now 
* now check the `With custom title` demo it at the top
* check the above things with the styling, location, buttons, etc.

| default header behaviour  | with custom title passed through |
| ------------- | ------------- |
| image  | image  |

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/components/date-range/style.scss [hunk 1]
```diff
@@ -248,40 +255,36 @@ $date-range-mobile-layout-switch: $break-small;
...
 
-	.DayPicker-Day--range .date-picker__day {
-		color: var(--color-text);
-		background-color: var(--date-range-picker-highlight-color);
-	}
-
-	.DayPicker-Day--start.DayPicker-Day--range-start,
-	.DayPicker-Day--end.DayPicker-Day--range-end {
-		& .date-picker__day {
+	.DayPicker-Day--range,
```
Comment (kangzj) [inline client/components/date-range/style.scss]: are these changes intended?

Comment (annacmc) [reply client/components/date-range/style.scss]: hmmm maybe my rebase got a bit out of control here. Those look like your changes from yesterday with the range styling, right?

Comment (kangzj) [reply client/components/date-range/style.scss]: Looks like a rebase issue here yes. If you take a look at the highlight colors, is it still working?

Comment (annacmc) [reply client/components/date-range/style.scss]: Fixing that up to bring all your work back in now. Sorry about that 🙈

File: client/components/date-range/header.tsx [hunk 2]
```diff
@@ -1,47 +1,73 @@
-import { Button } from '@automattic/components';
+import { Gridicon } from '@automattic/components';
+import { Button } from '@wordpress/components';
```
Comment (kangzj) [inline client/components/date-range/header.tsx]: Button import changed which caused the TS error?

**Review by kangzj — APPROVED** [review]
Nice refactoring! LGTM 🚀

File: client/components/date-range/header.tsx [hunk 3]
```diff
@@ -1,47 +1,72 @@
...
-				className="date-range__apply-btn"
-				onClick={ onApplyClick }
-				primary
-				compact
-				aria-label={ applyText }
-			>
-				{ applyText }
-			</Button>
+			<div className="date-range__controls">
+				{ customTitle ? (
```
Comment (kangzj) [inline client/components/date-range/header.tsx:64]: This is definitely isn't a blocker, but I'm still thinking maybe we shouldn't hide the help functions when there is a custom title, i.e. it'll be better if the custom title and the helpers could co-exist.

Just something for you to consider. Not a priority right now.

Great work!

Comment (annacmc) [reply client/components/date-range/header.tsx:64]: Thanks, I agree there may be some way for them to work together in a future iteration. We will figure something out!
